syntax = "proto3";

package servidor;

service opcoesCliente {
  rpc cadastrarCliente (Cliente) returns (Resposta) {}

  rpc checarCliente (Cliente) returns (Resposta) {}
}

service opcoesVoucher {
  rpc cadastrarVoucher (Voucher) returns (Resposta) {}

  rpc apresentarVouchersUsuario (ID) returns (Vouchers) {}

  rpc apresentarVouchers (ID) returns (Vouchers) {}
}

service opcoesTroca {
  rpc apresentarTrocas (Troca) returns (Trocas) {}

  rpc proporTroca (Troca) returns (Resposta) {}

  rpc realizarTroca (Troca) returns (Resposta) {}

  rpc negarTroca (Troca) returns (Resposta) {}
}

message Cliente {
  string nome = 1;
  string email = 2;
  string senha = 3;
}

message Voucher {
  int32 id = 1;
  string titulo = 2;
  string descricao = 3;
  string gato = 4;
  string local = 5;
  string lanche = 6;
  int32 duracao = 7;
  string imagem = 8;
  int32 titular_id = 9;
}

message Vouchers {
  repeated Voucher v = 1;
}

message Comp_Troca {
  int32 id = 1;
  Voucher v1 = 2;
  Voucher v2 = 3;
}

message Trocas {
  repeated Comp_Troca t = 1;
}

message Troca {
  int32 id1 = 1;
  int32 id2 = 2;
}

message Resposta {
  int32 message = 1;
}

message ID {
  int32 id = 1;
}